### PR TITLE
Poisson sampling transform for relations

### PR DESCRIPTION
**Pull request by andicuko**
Implement `Relation` with `poisson_sampling(proba: f64)` transform.

### CODE REVIEW

File: src/relation/transforms.rs [hunk 1]
```diff
@@ -701,6 +702,28 @@ impl Relation {
...
+        } else {
+            proba
+        };
+        let sampled_relation: Relation = Relation::map()
+            .with_iter(
+                self.schema()
+                    .iter()
+                    .map(|f| (f.name(), Expr::col(f.name()))),
+            )
+            .filter(Expr::lt(Expr::random(0), Expr::val(proba)))
```
Comment (ngrislain) [inline src/relation/transforms.rs:722]: Use: `namer::new_id("POISSON_SAMLPING")` and `reset` in tests

**Comment by vserret** [comment]
Maybe you should check that you transform is correctly transformed to an ast then in a string

**Comment by andicuko** [comment]
> Maybe you should check that you transform is correctly transformed to an ast then in a string

Ok. I added a test to check the query from the transform

File: src/relation/transforms.rs [hunk 2]
```diff
@@ -701,6 +702,31 @@ impl Relation {
             .input(self)
             .build()
     }
+
+    /// Poisson sampling of a relation. It samples each line with probability 0 <= proba <= 1
+    pub fn poisson_sampling(self, proba: f64) -> Relation {
+        //make sure proba is between 0 and 1.
+        let proba = if proba >= 1.0 {
```
Comment (ngrislain) [inline src/relation/transforms.rs:709]: use https://doc.rust-lang.org/std/primitive.f64.html#method.clamp instead

Comment (ngrislain) [reply src/relation/transforms.rs:709]: Assert this instead

**Pull request merged**